mlflow/server/handlers.py

```diff
@@ -1103,6 +1112,117 @@ def _default_history_bulk_impl():
+    num_steps = end_step - start_step + 1
+    interval = num_steps / max_results
+    grouped_steps = {}
+    # include both start_step & end_step
+    for i in range(start_step, end_step + 1):
+        idx = (i - start_step) // interval
+        if idx not in grouped_steps:
+            grouped_steps[idx] = i
+        else:
+            grouped_steps[idx] = min(grouped_steps[idx], i)
+    return list(grouped_steps.values())
+
+
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def get_metric_history_bulk_interval_handler():
+    MAX_RUNS_GET_METRIC_HISTORY_BULK = 100
+    MAX_RESULTS_PER_RUN = 2500
+    MAX_RESULTS_GET_METRIC_HISTORY = 25000
+
+    request_message = _get_request_message(
+        GetMetricHistoryBulkInterval(),
+        schema={
+            "run_ids": [
+                _assert_required,
+                _assert_array,
+                _assert_item_type_string,
+                lambda x: _assert_less_than_or_equal(
+                    len(x),
+                    MAX_RUNS_GET_METRIC_HISTORY_BULK,
+                    message=f"GetMetricHistoryBulkInterval request must specify at most "
+                    f"{MAX_RUNS_GET_METRIC_HISTORY_BULK} run_ids. Received {len(x)} run_ids.",
+                ),
+            ],
+            "metric_key": [_assert_required, _assert_string],
+            "start_step": [_assert_intlike],
+            "end_step": [_assert_intlike],
+            "max_results": [
+                _assert_intlike,
+                lambda x: _assert_intlike_within_range(
+                    int(x),
+                    1,
+                    MAX_RESULTS_PER_RUN,
+                    message=f"max_results must be between 1 and {MAX_RESULTS_PER_RUN}.",
+                ),
+            ],
+        },
+    )
+
+    args = request.args
+    run_ids = request_message.run_ids
+    metric_key = request_message.metric_key
+    max_results = int(args.get("max_results", MAX_RESULTS_PER_RUN))
+
+    store = _get_tracking_store()
+
+    def _get_max_step_for_metric(run_id, metric_key):
+        if hasattr(store, "get_max_step_for_metric"):
+            return store.get_max_step_for_metric(run_id=run_id, metric_key=metric_key)
+        steps = [m.step for m in store.get_metric_history(run_id, metric_key)]
+        return max(steps) if steps else 0
+
+    def _get_sampled_steps(run_ids, metric_key, max_results):
+        # can not fetch from request_message as the default value is 0
+        start_step = args.get("start_step")
+        end_step = args.get("end_step")
+        if start_step is None and end_step is None:
+            max_steps_per_run = [_get_max_step_for_metric(run_id, metric_key) for run_id in run_ids]
+            max_steps = max(max_steps_per_run)
+            # Get sampled steps and append the max step for each run.
+            sampled_steps = _get_sampled_steps_from_steps(0, max_steps, max_results)
+            return list(set(sampled_steps).union(max_steps_per_run))
```

Comment:
```suggestion
            return set(sampled_steps).union(max_steps_per_run)
```

Does this work?

Reply:
Nope, sqlalchemy store requires steps to be list